pkg/kubelet/apis/config/validation/validation_test.go

```diff
@@ -70,186 +71,454 @@ func TestValidateKubeletConfiguration(t *testing.T) {
 			Format: "text",
 		},
 	}
-	if allErrors := ValidateKubeletConfiguration(successCase1); allErrors != nil {
-		t.Errorf("expect no errors, got %v", allErrors)
-	}
+)
 
-	successCase2 := &kubeletconfig.KubeletConfiguration{
-		CgroupsPerQOS:                   true,
-		EnforceNodeAllocatable:          []string{"pods"},
-		SystemReservedCgroup:            "",
-		KubeReservedCgroup:              "",
-		SystemCgroups:                   "",
-		CgroupRoot:                      "",
-		EventBurst:                      10,
-		EventRecordQPS:                  5,
-		HealthzPort:                     10248,
-		ImageGCHighThresholdPercent:     85,
-		ImageGCLowThresholdPercent:      80,
-		IPTablesDropBit:                 15,
-		IPTablesMasqueradeBit:           14,
-		KubeAPIBurst:                    10,
-		KubeAPIQPS:                      5,
-		MaxOpenFiles:                    1000000,
-		MaxPods:                         110,
-		OOMScoreAdj:                     -999,
-		PodsPerCore:                     100,
-		Port:                            65535,
-		ReadOnlyPort:                    0,
-		RegistryBurst:                   10,
-		RegistryPullQPS:                 5,
-		HairpinMode:                     kubeletconfig.PromiscuousBridge,
-		NodeLeaseDurationSeconds:        1,
-		CPUCFSQuotaPeriod:               metav1.Duration{Duration: 50 * time.Millisecond},
-		ReservedSystemCPUs:              "0-3",
-		TopologyManagerScope:            kubeletconfig.ContainerTopologyManagerScope,
-		TopologyManagerPolicy:           kubeletconfig.NoneTopologyManagerPolicy,
-		ShutdownGracePeriod:             metav1.Duration{Duration: 10 * time.Minute},
-		ShutdownGracePeriodCriticalPods: metav1.Duration{Duration: 0},
-		MemoryThrottlingFactor:          utilpointer.Float64Ptr(0.9),
-		FeatureGates: map[string]bool{
-			"CustomCPUCFSQuotaPeriod": true,
-			"MemoryQoS":               true,
-		},
-		Logging: componentbaseconfig.LoggingConfiguration{
-			Format: "text",
+func TestValidateKubeletConfiguration(t *testing.T) {
+	successCases := []struct {
+		name      string
+		configure func(config *kubeletconfig.KubeletConfiguration) *kubeletconfig.KubeletConfiguration
+	}{
+		{
+			name: "Success",
+			configure: func(conf *kubeletconfig.KubeletConfiguration) *kubeletconfig.KubeletConfiguration {
+				return conf
+			},
 		},
 	}
-	if allErrors := ValidateKubeletConfiguration(successCase2); allErrors != nil {
-		t.Errorf("expect no errors, got %v", allErrors)
+
+	for _, tc := range successCases {
```

Comment:
(non-blocking) Rather than splitting success and error cases, we could have a single loop and the success case simply has an empty errMsg.